flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/ResultPartitionTest.java

```diff
@@ -360,6 +361,32 @@ private void testReleaseMemory(final ResultPartitionType resultPartitionType) th
 		}
 	}
 
+	/**
+	 * Tests {@link ResultPartition#getAvailableFuture()}.
+	 */
+	@Test
+	public void testIsAvailableOrNot() throws IOException, InterruptedException {
```

Comment:
Is this test adding a test coverage for something new? Or a missing coverage  for `getAvailableFuture()`? If only the latter, we could go away without this test, as the logic is almost entirely covered by `LocalBufferPoolTest`, but I'm also fine with adding some basic coverage here.

Reply:
I add this because it may help us verify the changes on `ResultPartitionWriter` (though most of them are API changes). I'll keep this. :)